### PR TITLE
fix(pandas): make case work for non-RangeIndex dataframes

### PR DESCRIPTION
**Pull request by dlovell**
## Description of changes

This PR makes `PandasExecutor` create the `Series` with an index that matches the incoming data. Currently, when the incoming data does not use a `RangeIndex`, the output index is a union of a `RangeIndex` and the incoming data index.

### CODE REVIEW

File: ibis/backends/pandas/tests/test_operations.py [hunk 1]
```diff
@@ -741,6 +741,30 @@ def test_simple_case_column(batting, batting_df):
...
+                (1, "one"),
+                (2, "two"),
+            ),
+            default="unk",
+        )
+
+    df = pd.DataFrame(
+        {
+            "A": pd.Series({i: i % 3 for i in (0, 1, 2, 4)}),
+            "B": 0,
```
[the remark this answers is not in the document]
Comment (dlovell) [reply ibis/backends/pandas/tests/test_operations.py:757]: `B` is required to have "the incoming data index" be misaligned with the `RangeIndex` that is created by invoking `pd.Series` without an index argument.

**Comment by cpcloud** [comment]
What's the use case that's enabled here that requires this change? What can you not do with the current codebase?

**Comment by dlovell** [comment]
> What's the use case that's enabled here that requires this change? What can you not do with the current codebase?

register a dataframe with non-range index and have cases work

**Comment by dlovell** [comment]
alternatively, should dataframes' index be sanitized on registration? is there a specification of what should be true about dataframes that are registered?

**Comment by cpcloud** [comment]
> register a dataframe with non-range index and have cases work

It would be good to have a less abstract example documented in this PR. Doesn't really even have to be code, just _some_ description that helps justify why we should take on any additional code to the pandas backend.

**Comment by dlovell** [comment]
> > register a dataframe with non-range index and have cases work
> 
> It would be good to have a less abstract example documented in this PR. Doesn't really even have to be code, just _some_ description that helps justify why we should take on any additional code to the pandas backend.

Does this match what you're looking for?

When I run this code
```
df = pd.DataFrame({
    "A": pd.Series({i: i % 3 for i in (0, 1, 2, 4)}),
    "B": 0,
})
expr = ibis.pandas.connect({"t": df}).table("t")


print("Input")
print(len(expr.execute()))
print(expr.execute())
print()


print("Current results")
x = expr.mutate(**{"A": lambda t: t["A"].pipe(do_replace)}).execute()
print(len(x))
print(x)
print()
```

I get these results
```
Input
4
   A  B
0  0  0
1  1  0
2  2  0
4  1  0

Current results
5
     A    B
0  unk  0.0
1  one  0.0
2  two  0.0
3  one  NaN
4  NaN  0.0
```

**Comment by cpcloud** [comment]
Heh, yeah, that does :)

I'll just keep suggesting that the pandas backend should probably be avoided.

I somewhat reluctantly will accept this PR, fully realizing that creating the pandas backend was probably a doomed idea from the start 😂

File: ibis/backends/pandas/tests/test_operations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/pandas/tests/test_operations.py:757]: Ahhh, ok, roger that.

**Pull request merged**